precompiles/gov/tx.go

```diff
@@ -16,12 +16,49 @@ import (
 )
 
 const (
+	// SubmitProposalMethod defines the ABI method name for the gov SubmitProposal transaction.
+	SubmitProposalMethod = "submitProposal"
 	// VoteMethod defines the ABI method name for the gov Vote transaction.
 	VoteMethod = "vote"
 	// VoteWeightedMethod defines the ABI method name for the gov VoteWeighted transaction.
 	VoteWeightedMethod = "voteWeighted"
 )
 
+
+// SubmitProposal defines a method to create new proposal given the messages.
+func (p Precompile) SubmitProposal(
+	ctx sdk.Context,
+	origin common.Address,
+	contract *vm.Contract,
+	stateDB vm.StateDB,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	msg, proposerHexAddr, err := NewMsgSubmitProposal(args)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the contract is the proposer, we don't need an origin check
+	// Otherwise check if the origin matches the proposer address
+	isContractVoter := contract.CallerAddress == proposerHexAddr && contract.CallerAddress != origin
+	if !isContractVoter && origin != proposerHexAddr {
+		return nil, fmt.Errorf(ErrDifferentOrigin, origin.String(), proposerHexAddr.String())
+	}
+
+	msgSrv := govkeeper.NewMsgServerImpl(&p.govKeeper)
+	res, err := msgSrv.SubmitProposal(ctx, msg)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
In case it is a call to this func within a contract call, and the deposit denom == evm denom, we have to make sure to add the corresponding balanceChange entry on the stateDB.

Another option could be to limit the call of this function for EOA only. So we don't have to deal with this handling for now